sds_data_manager/lambda_code/SDSCode/create_schema.py

```diff
@@ -13,18 +13,21 @@
 
 
 def send_response(event, context, response_status):
-    """Construct a response to indicate the status of the custom
-    resource lambda "SUCCESS" or "FAILED"
+    """Send the response.
+
+    Constructs a response to indicate the status of the custom resource lambda
+    "SUCCESS" or "FAILED"
 
     Parameters
-    -------
+    ----------
     event: dict
         JSON-formatted document that contains data for a Lambda function to process.
     context: Context
         provides methods and properties that provide information about the invocation,
         function, and execution environment.
-    reponse_status: str
+    response_status: str
```

Comment:
Looks like there is! https://numpydoc.readthedocs.io/en/latest/validation.html#docstring-validation-using-python

And it looks like there is indeed a specific check for spaces around colons (PR10).

Might be worth enabling this and choosing which checks we care about, but probably best of a separate PR.

Reply:
Here is an example of the default output:

```
(sds-data-manager-py3.9) mabo8927: python -m numpydoc.hooks.validate_docstrings create_schema.py 
+---------------------+------------------------------+---------+----------------------------------------------------+
| file                | item                         | check   | description                                        |
+=====================+==============================+=========+====================================================+
| create_schema.py:1  | create_schema                | ES01    | No extended summary found                          |
+---------------------+------------------------------+---------+----------------------------------------------------+
| create_schema.py:1  | create_schema                | SA01    | See Also section not found                         |
+---------------------+------------------------------+---------+----------------------------------------------------+
| create_schema.py:1  | create_schema                | EX01    | No examples section found                          |
+---------------------+------------------------------+---------+----------------------------------------------------+
| create_schema.py:15 | create_schema.send_response  | GL01    | Docstring text (summary) should start in the line  |
|                     |                              |         | immediately after the opening quotes (not in the   |
|                     |                              |         | same line, or leaving a blank line in between)     |
+---------------------+------------------------------+---------+----------------------------------------------------+
| create_schema.py:15 | create_schema.send_response  | GL02    | Closing quotes should be placed in the line after  |
|                     |                              |         | the last text in the docstring (do not close the   |
|                     |                              |         | quotes in the same line as the text, or leave a    |
|                     |                              |         | blank line between the last text and the quotes)   |
+---------------------+------------------------------+---------+----------------------------------------------------+
| create_schema.py:15 | create_schema.send_response  | GL03    | Double line break found; please use only one blank |
|                     |                              |         | line to separate sections or paragraphs, and do    |
|                     |                              |         | not leave blank lines at the end of docstrings     |
+---------------------+------------------------------+---------+----------------------------------------------------+
| create_schema.py:15 | create_schema.send_response  | PR01    | Parameters {'event', 'context', 'response_status'} |
|                     |                              |         | not documented                                     |
+---------------------+------------------------------+---------+----------------------------------------------------+
| create_schema.py:15 | create_schema.send_response  | PR02    | Unknown parameters {'event: dict', 'context:       |
|                     |                              |         | Context', 'response_status: str'}                  |
+---------------------+------------------------------+---------+----------------------------------------------------+
| create_schema.py:15 | create_schema.send_response  | PR10    | Parameter "event" requires a space before the      |
|                     |                              |         | colon separating the parameter name and type       |
+---------------------+------------------------------+---------+----------------------------------------------------+
| create_schema.py:15 | create_schema.send_response  | PR10    | Parameter "context" requires a space before the    |
|                     |                              |         | colon separating the parameter name and type       |
+---------------------+------------------------------+---------+----------------------------------------------------+
| create_schema.py:15 | create_schema.send_response  | PR08    | Parameter "context: Context" description should    |
|                     |                              |         | start with a capital letter                        |
+---------------------+------------------------------+---------+----------------------------------------------------+
| create_schema.py:15 | create_schema.send_response  | PR10    | Parameter "response_status" requires a space       |
|                     |                              |         | before the colon separating the parameter name and |
|                     |                              |         | type                                               |
+---------------------+------------------------------+---------+----------------------------------------------------+
| create_schema.py:15 | create_schema.send_response  | SA01    | See Also section not found                         |
+---------------------+------------------------------+---------+----------------------------------------------------+
| create_schema.py:15 | create_schema.send_response  | EX01    | No examples section found                          |
+---------------------+------------------------------+---------+----------------------------------------------------+
| create_schema.py:55 | create_schema.lambda_handler | ES01    | No extended summary found                          |
+---------------------+------------------------------+---------+----------------------------------------------------+
| create_schema.py:55 | create_schema.lambda_handler | PR01    | Parameters {'event', 'context'} not documented     |
+---------------------+------------------------------+---------+----------------------------------------------------+
| create_schema.py:55 | create_schema.lambda_handler | SA01    | See Also section not found                         |
+---------------------+------------------------------+---------+----------------------------------------------------+
| create_schema.py:55 | create_schema.lambda_handler | EX01    | No examples section found                          |
+---------------------+------------------------------+---------+----------------------------------------------------+

```

Reply:
Nice! I agree save it for a follow-up PR no problem. But that is nice to know because I think that may be causing some of our current doc build issues too, so maybe this will help there.